v2/client/client.go

```diff
@@ -200,27 +209,35 @@ func (c *ceClient) StartReceiver(ctx context.Context, fn interface{}) error {
 
 	var msg binding.Message
 	var respFn protocol.ResponseFn
-	// Start Polling.
-	for {
-		if c.responder != nil {
-			msg, respFn, err = c.responder.Respond(ctx)
-		} else if c.receiver != nil {
-			msg, err = c.receiver.Receive(ctx)
-		} else {
-			return errors.New("responder nor receiver set")
-		}
-
-		if err == io.EOF { // Normal close
-			return nil
-		}
-
-		if err != nil {
-			cecontext.LoggerFrom(ctx).Warnf("Error while receiving a message: %s", err)
-			continue
-		}
 
-		if err := c.invoker.Invoke(ctx, msg, respFn); err != nil {
-			return err
-		}
+	// Start Polling.
+	wg := sync.WaitGroup{}
+	for i := 0; i < c.pollGoroutines; i++ {
```

Comment:
I do not think you want to `i++` here, this looks like it will grow too large for long lived processes.

Reply:
?!

Reply:
nevermind!